sklearn/preprocessing/tests/test_data.py

```diff
@@ -2377,3 +2378,190 @@ def test_power_transformer_lambda_zero():
+    X2 = np.array([[0, 2, 1], [0, 2, 3], [1, 0, 5], [1, 1, 5]])
+    cf_inclusion_1 = CountFeaturizer(inclusion=[0])
+    cf_inclusion_2 = CountFeaturizer(inclusion=[0, 1, 2])
+    cf_inclusion_3 = CountFeaturizer(inclusion=[0, 1])
+    assert_array_equal(
+        cf_inclusion_1.fit_transform(X),
+        np.array([[0, 2, 1, 1], [1, 0, 3, 3], [1, 0, 2, 3], [1, 0, 2, 3]]))
+
+    assert_array_equal(
+        cf_inclusion_2.fit_transform(X),
+        np.array([[0, 2, 1, 1], [1, 0, 3, 1], [1, 0, 2, 2], [1, 0, 2, 2]]))
+
+    assert_array_equal(
+        cf_inclusion_3.fit_transform(X2),
+        np.array([[0, 2, 1, 2], [0, 2, 3, 2], [1, 0, 5, 1], [1, 1, 5, 1]]))
+
+
+def test_count_featurizer_y():
+    # test with the y parameter set
+
+    X = np.array([[0, 2, 1], [1, 0, 3], [1, 0, 2], [1, 0, 2]])
+    y = np.array([0, 0, 1, 0])
+    cf_1 = CountFeaturizer()
+    assert_array_equal(
+        cf_1.fit_transform(X, y=y),
+        np.array([[0, 2, 1, 1, 0], [1, 0, 3, 1, 0],
+                 [1, 0, 2, 1, 1], [1, 0, 2, 1, 1]]))
+
+
+def test_count_featurizer_y_inclusion():
+    # test with the inclusion and y parameter set
+
+    X = np.array([[0, 2, 1], [1, 0, 3], [1, 0, 2], [1, 0, 2]])
+    y = np.array([0, 0, 1, 0])
+    cf_inclusion_1 = CountFeaturizer(inclusion=[0])
+    cf_inclusion_2 = CountFeaturizer(inclusion=[0, 1, 2])
+    assert_array_equal(
+        cf_inclusion_1.fit_transform(X, y=y),
+        np.array([[0, 2, 1, 1, 0], [1, 0, 3, 2, 1],
+                 [1, 0, 2, 2, 1], [1, 0, 2, 2, 1]]))
+
+    assert_array_equal(
+        cf_inclusion_2.fit_transform(X, y=y),
+        np.array([[0, 2, 1, 1, 0], [1, 0, 3, 1, 0],
+                 [1, 0, 2, 1, 1], [1, 0, 2, 1, 1]]))
+
+
+def test_count_featurizer_multi_y():
+    # test with multi-dimensional y parameter
+
+    X = np.array([[0], [0], [1], [1]])
+    y = np.array([[0, 1], [0, 1], [0, 0], [0, 1]])
+    cf = CountFeaturizer()
+    assert_array_equal(
+        cf.fit_transform(X, y=y),
+        np.array([[0, 2, 0, 2], [0, 2, 0, 2],
+                 [1, 2, 1, 1], [1, 2, 1, 1]]))
+
+
+def test_count_featurizer_multi_inclusion():
+    # test with 2D inclusion
+
+    X = np.array([[0, 0], [0, 0], [1, 0], [1, 0]])
+    y = np.array([[0], [0], [0], [1]])
+    cf = CountFeaturizer(inclusion=np.array([[0], [1]]))
+    assert_array_equal(
+        cf.fit_transform(X, y=y),
+        np.array([[0, 0, 2, 0, 3, 1], [0, 0, 2, 0, 3, 1],
+                 [1, 0, 1, 1, 3, 1], [1, 0, 1, 1, 3, 1]]))
+
+    # each should be the same as [[0], [1]]
+    cf = CountFeaturizer(inclusion="each")
+    assert_array_equal(
+        cf.fit_transform(X, y=y),
+        np.array([[0, 0, 2, 0, 3, 1], [0, 0, 2, 0, 3, 1],
+                 [1, 0, 1, 1, 3, 1], [1, 0, 1, 1, 3, 1]]))
+
+    cf = CountFeaturizer(inclusion=[[0]])
+    assert_array_equal(
+        cf.fit_transform(X, y=y),
+        np.array([[0, 0, 2, 0], [0, 0, 2, 0],
+                 [1, 0, 1, 1], [1, 0, 1, 1]]))
+
+    cf = CountFeaturizer(inclusion="each")
```

Comment:
* test_count_featurizer_multi_inclusion line 2527: Please explain how this results in 12 new outputs

Reply:
Each column of y gets its own columns outputted for each unique value it can take. For example, here, the first column of y gets 2 unique values (0 and 1), so 2 columns are outputted for each inclusion list (since 'each' is used, there are 2 inclusion lists [0] and [1]). The second column of y gets 4 unique values (0, 1, 2, 3) so 4 columns are outputted for each inclusion list. As a result (4 + 2, the total # of different y values) x (2, the size of the inclusion list) = 12 columns are outputted